temporalio/worker/_worker.py

```diff
@@ -48,6 +49,48 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
```

Comment:
Would recommend these be set to frozen, especially this one with how it's used as a default in a parameter (Python parameter defaults are a bit strange in that that expression is evaluated once and reused for every default).